solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2ClusterStateProvider.java

```diff
@@ -19,33 +19,34 @@
 
 import java.io.IOException;
 import java.util.List;
-import org.apache.solr.client.solrj.SolrClient;
 
-public class Http2ClusterStateProvider extends BaseHttpClusterStateProvider {
-  final Http2SolrClient httpClient;
-  final boolean closeClient;
+public class Http2ClusterStateProvider<C extends HttpSolrClientBase>
+    extends BaseHttpClusterStateProvider {
+  final C httpClient;
 
-  public Http2ClusterStateProvider(List<String> solrUrls, Http2SolrClient httpClient)
-      throws Exception {
-    this.httpClient = httpClient == null ? new Http2SolrClient.Builder().build() : httpClient;
-    this.closeClient = httpClient == null;
+  public Http2ClusterStateProvider(List<String> solrUrls, C httpClient) throws Exception {
```

Comment:
Should document that we'll always close the client.  Arguably we shouldn't if the caller isn't providing it, as that's the general rule of responsibility for closing things (creator closes).